var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -126,13 +126,20 @@ class Boost(Package):
     depends_on('mpi', when='+mpi')
     depends_on('bzip2', when='+iostreams')
     depends_on('zlib', when='+iostreams')
+    depends_on('zlib', when='%xl_r' )
```

Comment:
At the moment, those constrains are easier to specify in `packages.yaml` as opposed to CLI. http://spack.readthedocs.io/en/latest/configuration.html